### PR TITLE
Add a FakeTransactionalJobQueue and bind it with FakeJobQueueModule

### PR DESCRIPTION
**Pull request by tso**

### CODE REVIEW

File: misk-jobqueue-testing/src/main/kotlin/misk/jobqueue/FakeTransactionalJobQueue.kt [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+ * ```
+ * callApiThatEnqueuesJob()
+ * assertJobIsCorrect(fakeJobQueue.peekJobs(MY_QUEUE).first())
+ *
+ * fakeJobQueue.handleJobs()
+ * assertJobSideEffects()
+ * ```
+ */
+@Singleton
+class FakeTransactionalJobQueue @Inject constructor(
```
Comment (ryanhall07) [inline misk-jobqueue-testing/src/main/kotlin/misk/jobqueue/FakeTransactionalJobQueue.kt]: I think we should get in the practice of writing tests for our fakes, if they are public fakes that we expect devs to use. In an ideal world, the test would also verify the fake behaves like the real, but that's probably overkill for now.

Comment (tso) [reply misk-jobqueue-testing/src/main/kotlin/misk/jobqueue/FakeTransactionalJobQueue.kt]: Yep, definitely. I'll add some tests

File: build.gradle [hunk 2]
```diff
@@ -104,7 +104,7 @@ subprojects { subProject ->
     apply from: rootProject.file("hooks.gradle")
   }
 
-  if (["misk-aws","misk-events","misk-jobqueue","misk-hibernate","misk-hibernate-testing"].contains(name)) {
+  if (["misk-aws","misk-events","misk-jobqueue","misk-jobqueue-testing","misk-hibernate","misk-hibernate-testing"].contains(name)) {
```
Comment (swankjesse) [inline build.gradle]: wow

File: misk-jobqueue-testing/src/main/kotlin/misk/jobqueue/FakeTransactionalJobQueue.kt [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+ * trigger processing.
+ *
+ * Example usage might look like this:
+ * ```
+ * callApiThatEnqueuesJob()
+ * assertJobIsCorrect(fakeJobQueue.peekJobs(MY_QUEUE).first())
+ *
+ * fakeJobQueue.handleJobs()
+ * assertJobSideEffects()
+ * ```
```
Comment (swankjesse) [inline misk-jobqueue-testing/src/main/kotlin/misk/jobqueue/FakeTransactionalJobQueue.kt]: This doc is SO good

File: misk-jobqueue-testing/src/main/kotlin/misk/jobqueue/FakeTransactionalJobQueue.kt [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    val jobs = jobQueues[queueName] ?: return
+
+    while (true) {
+      val job = jobs.poll() ?: break
+      jobHandler.handleJob(job)
+      check(job.acknowledged) { "Expected $job to be acknowledged after handling" }
+    }
+  }
+
+  fun handleJobs() = jobQueues.keys.forEach { handleJobs(it) }
```
Comment (swankjesse) [inline misk-jobqueue-testing/src/main/kotlin/misk/jobqueue/FakeTransactionalJobQueue.kt]: brilliant

File: misk-jobqueue-testing/src/test/kotlin/misk/jobqueue/FakeTransactionalJobQueueTest.kt [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+      unitEnqueuer.enqueue(session, Unit.LIBERATOR, STARPORT_QUEUE)
+    }
+
+    val factoryJobs = fakeTransactionalJobQueue.peekJobs(FACTORY_QUEUE)
+    assertThat(factoryJobs).hasSize(1)
+
+    val starportJobs = fakeTransactionalJobQueue.peekJobs(STARPORT_QUEUE)
+    assertThat(starportJobs).hasSize(2)
+
+    assertThat(factoryJobs[0].id).isEqualTo("fakej0bqee000000000000001")
```
Comment (swankjesse) [inline misk-jobqueue-testing/src/test/kotlin/misk/jobqueue/FakeTransactionalJobQueueTest.kt]: I’m so proud of this silly little fake token maker

**Pull request merged**